File: web/projects/portal/src/app/vsobjects/event/table/drill-cells-event.ts
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { BaseTableDrillEvent } from "./base-table-drill-event";
import { DrillEvent } from "./drill-event";

/**
 * Drill cells.
 */
export class DrillCellsEvent extends BaseTableDrillEvent {
   constructor(assemblyName: string, drillEvents: DrillEvent[], drillUp: boolean = false,
               private drillTarget: DrillTarget = DrillTarget.NONE, private field: string = null)
   {
      super(assemblyName, drillEvents, drillUp);
   }
}

export enum DrillTarget {
   CROSSTAB,
   FIELD,
   NONE
}

File: core/src/main/java/inetsoft/web/viewsheet/event/table/DrillCellsEvent.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.viewsheet.event.table;

public class DrillCellsEvent extends BaseTableDrillEvent {

   public DrillTarget getDrillTarget() {
      return drillTarget;
   }

   public void setDrillTarget(DrillTarget drillTarget) {
      this.drillTarget = drillTarget;
   }

   public String getField() {
      return field;
   }

   public void setField(String field) {
      this.field = field;
   }

   private DrillTarget drillTarget;
   private String field;

   public static enum DrillTarget {
      CROSSTAB,
      FIELD,
      NONE
   }
}
